### ISSUE TITLE
Example usage broken

### ISSUE THREAD

**Issue by MiiCode2** [catdad/gulp-each#6]
The callback in your examples used to work in 0.2.0.  This is no longer the case with your current code.  In order for your usage examples to work, we need to change

```callback(null, newContent);```

to

```
file.contents = new Buffer(newContent);
callback(null, file);
```

Otherwise you will experience this error:

```TypeError: Path must be a string. Received undefined```

Also, a minor fix for your examples

        }) // <-- add extra ) here
        .pipe(gulp.dest('output'));

        }, 'buffer') // <-- add extra ) here
        .pipe(gulp.dest('output'));

**Comment by catdad** [comment]
Hmm... yeah, you are right. The examples definitely have a typo. It used to work by accident in the previous version, but really, the examples have always broken the `gulp` spec.

**Comment by catdad** [comment]
I have published 0.2.1 and 0.3.1 with fixed examples. Thanks for the feedback.

**Comment by catdad** [comment]
Ugh... submitting 2 bugs in one issue is never a good idea. People miss the real bug.

I think you are suggesting that there is a bug with using different version of vinyl. Gulp seems to use an older version than the one used in `read-vinyl-file-stream`.

What version of `gulp` are you using?

**Comment by MiiCode2** [comment]
Sorry about the two bugs in one issue.

I am using version 3.9.1 of gulp.

**Comment by catdad** [comment]
Okay, so this is actually a bug in `read-vinyl-file-stream` that will take a bit more effort than I am willing to put in tonight.

I took the 0.2.x code and republished it as 0.4.0. I am also trying to deprecate 0.3.x, but it's not working (see https://github.com/npm/npm/issues/15619). For now, this should be enough. I will look into actually fixing the new code tomorrow.

**Comment by MiiCode2** [comment]
Thanks for looking into this.

I tried to track down the problem myself, but with all the refactoring you did going from 0.2.0 to 0.3.0, I decided it was best to make you aware of the problem.

**Comment by catdad** [comment]
Deprecating is a registry issue, it seems: https://github.com/npm/registry/issues/131

**Comment by catdad** [comment]
Okay. The `read-vinyl-file-stream` bug has been fixed in https://github.com/catdad/read-vinyl-file-stream/pull/2, followed by an update to this module along with an explicit `gulp` pipeline test added in #7.

I have now published this update to v0.5.0.

**Comment by MiiCode2** [comment]
Thanks I can confirm this is now working.
